include/rdma/fi_domain.h

```diff
@@ -102,6 +104,23 @@ struct fi_ops_av {
 			char *buf, size_t *len);
 	int	(*av_set)(struct fid_av *av, struct fi_av_set_attr *attr,
 			struct fid_av_set **av_set, void *context);
+	int	(*insert_auth_key)(struct fid_av *av, const void *auth_key,
+				   uint64_t flags, void *context);
+	int	(*lookup_auth_key)(struct fid_av *av, fi_addr_t addr,
+				   void *auth_key, size_t *auth_key_len);
```

Comment:
If we go with the above changes, then lookup would take the key_index as input, not the fi_addr_t.  Requiring an fi_addr_t value here means that we can't actually lookup an auth_key until after it has been associated with some address, which seems restrictive.  This suggests an API:
```
(*lookup_auth_key)(struct fid_av *av, int key_index, void *auth_key, size_t auth_key_size);
```

Reply:
Adding to the previous comment, having the ability to lookup the auth key given an fi_addr_t still seems useful.  But if we change lookup as suggested, then we either need another lookup call or a 2-step process.  I'm suggesting the latter:
```
(*lookup_key_index)(struct fid_av *av, fi_addr_t, int *key_index);
```
A lookup from an fi_addr_t would involve calling both lookup functions.

Reply:
I think being able to lookup an auth_key from a fi_addr_t is required for DAOS to use this. CQ event with FI_SOURCE will return fi_addr_t encoding the EP addr and auth_key. DAOS servers need to serialize the the EP addr and auth_key when forwarding the RPC to other servers.

Reply:
The point I was making is that with the proposal, there's isn't a way to lookup the auth_key *without* it being attached to an address.  The app has to keep the full auth_key around for later use.